hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -87,11 +91,14 @@ public DerbyDialect(int version) {
 		registerColumnType( Types.BIT, 1, "boolean" ); //no bit
 		registerColumnType( Types.BIT, "smallint" ); //no bit
 		registerColumnType( Types.TINYINT, "smallint" ); //no tinyint
+		registerColumnType( Types.CHAR, "char(1)" );
 
 		//HHH-12827: map them both to the same type to
 		//           avoid problems with schema update
 //		registerColumnType( Types.DECIMAL, "decimal($p,$s)" );
 		registerColumnType( Types.NUMERIC, "decimal($p,$s)" );
+		registerColumnType( Types.FLOAT, "float" );
+		registerColumnType( Types.DOUBLE, "double" );
```

Comment:
No, we're not using this mapping anymore in Hibernate 6, that was what we used to using in Hibernate 5 and it was wrong and I changed it across all the `Dialect`s.

I'm pretty sure you shouldn't need to be changing type mappings for Derby because I tested the shit out of all this on Derby when I cleaned up the dialects.

Reply:
See 

- https://db.apache.org/derby/docs/10.8/ref/rrefsqlj27281.html
- https://db.apache.org/derby/docs/10.8/ref/rrefsqljdoubleprecision.html

The default type mappings in `Dialect` should be perfect for Derby.

If you map `Types.FLOAT` to `float`, then Java `float`s and `doubles will both be mapped to the precision-less SQL type `float`.

Reply:
Ok, I just tried to copy over parts from 5 that seemed to be missing, but if you did this on purpose, I'll remove the double and float registrations.

Reply:
No no, please don't do that. Nothing is missing, it's all defined in the superclass now. If you just copy stuff over from Hibernate 5, it will be wrong.